### PR TITLE
LPS-132799 Search Location tests

### PR DESCRIPTION
**Pull request by rodrigopaulino**
Related Issue: https://issues.liferay.com/browse/LPS-132799

### CODE REVIEW

**Comment by liferay-forms** [comment]
ci:test:sf

**Comment by liferay-forms** [comment]
Hi, you don’t need to run the CI commands anymore, I’ll do it for you and let you know if any issue happens.

**Comment by liferay-forms** [comment]
ci:test:relevant

**Comment by rodrigopaulino** [comment]
@natocesarrego, there's a lot to cover here and, as you already know, it took a lot of time to complete it (not just because of the amount of cases covered, but also because I had to keep pausing and resuming the work due to other dependencies). So it's possible that decisions were made during the early stages of development that I stopped following later on because they didn't make sense to me anymore. Although I've tried to organize everything cohesively before creating this PR.
Having said that, please feel free to suggest any refactoring, because I do believe there might be something to improve and I could really use a new set of eyes to check it for me.

Thanks beforehand.

**Comment by liferay-forms** [comment]
Hey @rodrigopaulino, can you please check if these failures are related to your changes? Thanks

**Comment by rodrigopaulino** [comment]
The failure is related to portal-impl and I didn't change that.

**Comment by natocesarrego** [comment]
@rodrigopaulino, can you please rebase this pull to solve the conflicts?

Thanks in advance :slightly_smiling_face:

**Comment by rodrigopaulino** [comment]
> @rodrigopaulino, can you please rebase this pull to solve the conflicts?
> 
> Thanks in advance 🙂

Done

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/internal/search/location/SearchLocationDDMFormFieldValueRendererTest.java [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+				new SearchLocationDDMFormFieldValueRenderer();
+
+		Assert.assertEquals(
+			"{&#34;city&#34;:&#34;Recife&#34;}",
+			searchLocationDDMFormFieldValueRenderer.render(
+				ddmFormFieldValue, LocaleUtil.US));
+	}
+
+	@Test
+	public void testRender2WithCityValue() {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/internal/search/location/SearchLocationDDMFormFieldValueRendererTest.java:66]: @rodrigopaulino, let's rename to `testRenderWithDDMFormFieldNameAndCityValue`.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/internal/redirect/button/RedirectButtonDDMFormFieldTemplateContextContributorTest.java [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+			parameters.get("message"));
+		Assert.assertEquals(
+			"http//localhost/test?portletId_mvcRenderCommandName=" +
+				"mvcRenderCommandName;portletId_parameterName=parameterValue",
+			parameters.get("redirectURL"));
+		Assert.assertEquals("title", parameters.get("title"));
+	}
+
+	@Test
+	public void testGetParameters2() {
```
Comment (natocesarrego) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/internal/redirect/button/RedirectButtonDDMFormFieldTemplateContextContributorTest.java:103]: @rodrigopaulino, please rename to `testGetParametersWithoutMessageArgumentsAndParameters`.

**Comment by rodrigopaulino** [comment]
I'm on it, @natocesarrego. Thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/BaseDDMFormFieldTypeSettingsTestCase.java [hunk 3]
```diff
@@ -49,6 +62,28 @@ public void setUp() throws Exception {
...
+				actualObject1, expectedObjects1.remove(0),
+				expectedObjects2 -> actualObject2 -> _assertObject(
+					actualObject2, expectedObjects2.remove(0),
+					expectedObjects3 ->
+						actualObject3 -> _assertDDMFormLayoutColumn(
+							(DDMFormLayoutColumn)actualObject3,
+							(DDMFormLayoutColumn)expectedObjects3.remove(0)),
+					"getDDMFormLayoutColumns", null),
+				"getDDMFormLayoutRows", null),
+			"getDDMFormLayoutPages",
```
[the remark this answers is not in the document]
Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/BaseDDMFormFieldTypeSettingsTestCase.java]: Done.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/test/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/HasGooglePlacesAPIKeyFunctionTest.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+ */
+public class HasGooglePlacesAPIKeyFunctionTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayUnitTestRule liferayUnitTestRule =
+		LiferayUnitTestRule.INSTANCE;
+
+	@Test
+	public void testApplyFalse1() {
```
[the remark this answers is not in the document]
Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/test/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/HasGooglePlacesAPIKeyFunctionTest.java]: Done.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/test/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/HasGooglePlacesAPIKeyFunctionTest.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	@Test
+	public void testApplyFalse2() {
+		HasGooglePlacesAPIKeyFunction hasGooglePlacesAPIKeyFunction =
+			_createHasGooglePlacesAPIKeyFunction(() -> null);
+
+		Assert.assertFalse(hasGooglePlacesAPIKeyFunction.apply());
+	}
+
+	@Test
+	public void testApplyTrue() {
```
[the remark this answers is not in the document]
Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/test/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/function/HasGooglePlacesAPIKeyFunctionTest.java]: Done.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/internal/search/location/SearchLocationDDMFormFieldValueRendererTest.java (hunk 1 above)
Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/internal/search/location/SearchLocationDDMFormFieldValueRendererTest.java]: Done.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/internal/redirect/button/RedirectButtonDDMFormFieldTemplateContextContributorTest.java (hunk 2 above)
Comment (rodrigopaulino) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/test/java/com/liferay/dynamic/data/mapping/form/field/type/internal/redirect/button/RedirectButtonDDMFormFieldTemplateContextContributorTest.java]: Done.

**Comment by rodrigopaulino** [comment]
ci:forward

**Comment by rodrigopaulino** [comment]
@natocesarrego, FYI I've removed the builder classes. They turned out to be an overkill, indeed. Thanks for inciting the reflection about them.

**Comment by liferay-forms** [comment]
ci:test:relevant

**Pull request closed without merging**